### PR TITLE
New Recipe: isoband v0.2.2

### PR DESCRIPTION
**Pull request by jkrumbiegel**
This is my first attempt at using BinaryBuilder, so if someone experienced could take a look to see if what I did makes sense I'd appreciate it.

The code being built is very simple, it's just two or three C++ files without make script or anything like that. I create a shared object from that which I confirmed to work locally. Tests would live in the package that imports the jll, because originally tests were written in R, not C++

### CODE REVIEW

File: I/isoband/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/jkrumbiegel/isoband.git", "8c0050541c02d0cc5b3dd737c78cbf81ee61f802")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+mkdir ${libdir}
```
Comment (giordano) [inline I/isoband/build_tarballs.jl:16]: Generally speaking, I recommend doing
```suggestion
mkdir -p ${libdir}
```
to make sure you don't fail if the directory already exists

File: I/isoband/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+$CXX -shared -std=c++11 -O3 -fPIC -o ${libdir}/isoband isoband/src/isoband.cpp
```
Comment (giordano) [inline I/isoband/build_tarballs.jl:17]: You need to add the extension to the file.  Also, the name of the library file starts with `lib*`:
```suggestion
$CXX -shared -std=c++11 -O3 -fPIC -o ${libdir}/libisoband.${dlext} isoband/src/isoband.cpp
```
I believe `-std=c++11` requires GCC v5?

File: I/isoband/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
```
Comment (giordano) [inline I/isoband/build_tarballs.jl:22]: Quoting from the log:
```
┌ Warning: /agent/_work/1/s/I/isoband/build/x86_64-linux-gnu/5ZKKdyhP/destdir/lib/isoband contains std::string values!  This causes incompatibilities across the GCC 4/5 version boundary.  To remedy this, you must build a tarball for both GCC 4 and GCC 5.  To do this, immediately after your `platforms` definition in your `build_tarballs.jl` file, add the line:
│ 
│     platforms = expand_cxxstring_abis(platforms)
└ @ BinaryBuilder /depot/packages/BinaryBuilder/JlR0m/src/auditor/compiler_abi.jl:222
```

```suggestion
platforms = expand_cxxstring_abis(supported_platforms())
```

File: I/isoband/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("isoband", :isoband)
```
Comment (giordano) [inline I/isoband/build_tarballs.jl:26]: To match the name of the file I suggested above:
```suggestion
    LibraryProduct("libisoband", :libisoband)
```

**Comment by jkrumbiegel** [comment]
cool, that seems to have fixed everything :) I'm not sure about your gcc5 comment, as I have no idea what the differences are between the compilers

**Pull request merged**